ssl/quic/quic_port.c

```diff
@@ -136,9 +140,24 @@ static int port_init(QUIC_PORT *port)
     ossl_list_port_insert_tail(&port->engine->port_list, port);
     port->on_engine_list    = 1;
     port->bio_changed       = 1;
+
+    /* Generate random key for token encryption */
+    if ((ctx = EVP_CIPHER_CTX_new()) == NULL
+        || (cipher = EVP_CIPHER_fetch(port->engine->libctx,
+                                      "AES-256-GCM", NULL)) == NULL
+        || !EVP_EncryptInit_ex(ctx, cipher, NULL, NULL, NULL)
+        || (key_len = EVP_CIPHER_CTX_get_key_length(ctx)) == 0
+        || (port->token_key = OPENSSL_malloc(key_len)) == NULL
```

Comment:
Instead of storing the token key, it would be actually much more efficient to store pre-initialized EVP_CIPHER_CTX  contexts for encryption and decryption with the key already set in. Then during the actual encryption and decryption you just pass the iv to the EVP_Encrypt/DecryptInit and set the EVP_CIPHER and key to NULL.